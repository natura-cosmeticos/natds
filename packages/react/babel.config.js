const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
