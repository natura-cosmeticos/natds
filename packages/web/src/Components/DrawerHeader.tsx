import React, { FunctionComponent, forwardRef } from "react";
import { withTheme } from "@material-ui/core";
import styled from "styled-components";
import { IThemeWeb } from "Themes";
import { getDefaultTheme } from "./shared";

import Typography from "./Typography";
import Avatar from "./Avatar";

export interface IDrawerHeaderProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  primary?: React.ReactNode;
  secondary?: React.ReactNode;
  avatarSrc?: string;
  avatarChildren?: React.ReactNode;
  theme: IThemeWeb | unknown;
}

const buildContent = ({
  primary, secondary, avatarSrc, avatarChildren, theme,
}: IDrawerHeaderProps) => <>
  {(avatarSrc || avatarChildren) && <DrawerHeaderAvatar as={Avatar} src={avatarSrc} size="large" theme={theme}>{avatarChildren}</DrawerHeaderAvatar>}
  {primary && <Typography variant="h5">{primary}</Typography>}
  {secondary && <Typography variant="subtitle2" color="textSecondary">{secondary}</Typography>}
</>;

export const DrawerHeader: FunctionComponent<IDrawerHeaderProps> = forwardRef((
  props: IDrawerHeaderProps,
  ref: any,
) => {
  const {
    children, component, theme: providerTheme, ...rest
  } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);
  const content = children || buildContent({
    ...props,
    theme,
  });

  return (
    <DrawerHeaderComponent
      as={component}
      theme={theme}
      {...rest}
      ref={ref}>
      {content}
    </DrawerHeaderComponent>
  );
});

export default withTheme(DrawerHeader);

const DrawerHeaderComponent = styled.div<{ theme: IThemeWeb }>`
  border-bottom: ${({ theme }) => `${theme.palette.divider} 1px solid`};
  flex: 0 0 auto;
  padding: ${({ theme }) => `${theme.sizes.semi}px ${theme.sizes.small}px ${theme.sizes.small}px`};
`;

const DrawerHeaderAvatar = styled.div`
  margin-bottom: ${({ theme }) => `${theme.sizes.small}px`};
`;
