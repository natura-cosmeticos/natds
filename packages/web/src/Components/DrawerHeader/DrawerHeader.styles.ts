import { IThemeWeb } from '../../Themes'
import { createStyles } from '../../styles/createStyles'
import { makeStyles } from '../../styles/makeStyles'

const style = ({ palette, sizes }: IThemeWeb) => createStyles({
  root: {
    borderBottom: `${palette?.divider} 1px solid`,
    flex: '0 0 auto',
    padding: `${sizes?.semi}px ${sizes?.small}px ${sizes?.small}px`
  },
  avatar: {
    marginBottom: sizes?.small
  }
})

export const useStyles = makeStyles(style, { name: 'NatDSDrawerHeader' })
export default useStyles
