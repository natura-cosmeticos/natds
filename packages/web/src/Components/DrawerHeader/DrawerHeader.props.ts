import * as React from 'react'

export interface IDrawerHeaderProps {
  avatarChildren?: React.ReactNode;

  /**
   * The component used for the avatar node.
   *
   * Either a `string` to use or a HTML element or a component.
   *
   * Defaults to `Avatar`
   *
   * @optional
   * @type element
   */
  avatarComponent?: React.ElementType;
  avatarSrc?: string;
  children?: React.ReactNode;

  /**
   * The component used for the root node.
   *
   * Either a `string` to use or a HTML element or a component.
   *
   * @default "div"
   * @optional
   * @type element
   */
  component?: React.ElementType;
  primary?: React.ReactNode;
  secondary?: React.ReactNode;
}
