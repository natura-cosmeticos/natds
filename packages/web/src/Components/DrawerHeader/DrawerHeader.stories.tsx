// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { DrawerHeader, IDrawerHeaderProps } from './DrawerHeader'

export default {
  component: DrawerHeader,
  title: 'Components/Drawer/Drawer Header'
} as Meta

const Template : Story<IDrawerHeaderProps> = (args: IDrawerHeaderProps) => <DrawerHeader {...args} />

export const Playground : Story<IDrawerHeaderProps> = Template.bind({})
Playground.args = {
  avatarSrc: 'https://pbs.twimg.com/profile_images/1308179129267781632/wLDnEiOy_400x400.jpg',
  primary: 'Hello, Long Name Lorem ipsum dolor sit amet...',
  secondary: 'Option'
}
