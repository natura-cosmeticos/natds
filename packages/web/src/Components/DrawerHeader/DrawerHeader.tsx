// eslint-disable-next-line no-use-before-define
import * as React from 'react'

import Typography from '../Typography'
import Avatar from '../Avatar'
import { IDrawerHeaderProps } from './DrawerHeader.props'
import useStyles from './DrawerHeader.styles'

export { IDrawerHeaderProps } from './DrawerHeader.props'

/**
 * ## Importing
 *
 * ```
 * import { DrawerHeader } from '@naturacosmeticos/natds-web';
 * ```
 */
export const DrawerHeader = React.forwardRef<HTMLElement, IDrawerHeaderProps>((
  props: IDrawerHeaderProps, ref
) => {
  const {
    avatarChildren,
    avatarComponent: AvatarComponent = Avatar,
    avatarSrc,
    children,
    component: Component = 'div',
    primary,
    secondary,
    ...otherProps
  } = props

  const { avatar, root } = useStyles()

  let content = (
    <>
      {(avatarSrc || avatarChildren) && <AvatarComponent className={avatar} size="large" src={avatarSrc}>{avatarChildren}</AvatarComponent>}
      {primary && <Typography variant="h5">{primary}</Typography>}
      {secondary && <Typography variant="subtitle2" color="textSecondary">{secondary}</Typography>}
    </>
  )

  if (children) {
    content = <>{children}</>
  }

  return (
    <Component
      className={root}
      {...otherProps}
      ref={ref}
    >
      {content}
    </Component>
  )
})

DrawerHeader.displayName = 'DrawerHeader'

export default DrawerHeader
