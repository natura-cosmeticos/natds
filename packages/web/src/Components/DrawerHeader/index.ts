import DrawerHeader from './DrawerHeader'

export { IDrawerHeaderProps } from './DrawerHeader.props'
export { DrawerHeader }
export default DrawerHeader
