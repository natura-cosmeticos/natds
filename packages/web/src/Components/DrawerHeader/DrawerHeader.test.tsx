import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { DrawerHeader } from './DrawerHeader'

describe('DrawerHeader component', () => {
  it('should match snapshot - DrawerHeader Avatar Image', () => {
    const children = 'some_image'
    const component = TestRenderer.create(<DrawerHeader avatarSrc={children} />)

    expect(component).toMatchSnapshot('DrawerHeader with Image Avatar snapshot')
  })

  it('should match snapshot - DrawerHeader Avatar Text', () => {
    const children = 'some_text'
    const component = TestRenderer.create(<DrawerHeader avatarChildren={children} />)

    expect(component).toMatchSnapshot('DrawerHeader with Text Avatar snapshot')
  })

  it('should match snapshot - DrawerHeader Title', () => {
    const children = 'some_title'
    const component = TestRenderer.create(<DrawerHeader primary={children} />)

    expect(component).toMatchSnapshot('DrawerHeader with Title snapshot')
  })

  it('should match snapshot - DrawerHeader Subtitle', () => {
    const children = 'some_title'
    const component = TestRenderer.create(<DrawerHeader secondary={children} />)

    expect(component).toMatchSnapshot('DrawerHeader with Subtitle snapshot')
  })

  it('should match snapshot - DrawerHeader Custom children', () => {
    const children = <div>some custom children</div>
    const component = TestRenderer.create(<DrawerHeader>{children}</DrawerHeader>)

    expect(component).toMatchSnapshot('DrawerHeader with Custom children snapshot')
  })
})
