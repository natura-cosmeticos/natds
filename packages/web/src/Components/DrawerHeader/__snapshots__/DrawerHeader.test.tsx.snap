// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader component should match snapshot - DrawerHeader Avatar Image: DrawerHeader with Image Avatar snapshot 1`] = `
<div
  className="NatDSDrawerHeader-root-1"
>
  <div
    className="MuiAvatar-root NatDsAvatar-root-3 MuiAvatar-circular NatDSDrawerHeader-avatar-2"
  >
    <img
      className="MuiAvatar-img"
      src="some_image"
    />
  </div>
</div>
`;

exports[`DrawerHeader component should match snapshot - DrawerHeader Avatar Text: DrawerHeader with Text Avatar snapshot 1`] = `
<div
  className="NatDSDrawerHeader-root-1"
>
  <div
    className="MuiAvatar-root NatDsAvatar-root-5 MuiAvatar-circular NatDSDrawerHeader-avatar-2 MuiAvatar-colorDefault"
  >
    some_text
  </div>
</div>
`;

exports[`DrawerHeader component should match snapshot - DrawerHeader Custom children: DrawerHeader with Custom children snapshot 1`] = `
<div
  className="NatDSDrawerHeader-root-1"
>
  <div>
    some custom children
  </div>
</div>
`;

exports[`DrawerHeader component should match snapshot - DrawerHeader Subtitle: DrawerHeader with Subtitle snapshot 1`] = `
<div
  className="NatDSDrawerHeader-root-1"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
  >
    some_title
  </h6>
</div>
`;

exports[`DrawerHeader component should match snapshot - DrawerHeader Title: DrawerHeader with Title snapshot 1`] = `
<div
  className="NatDSDrawerHeader-root-1"
>
  <h5
    className="MuiTypography-root MuiTypography-h5"
  >
    some_title
  </h5>
</div>
`;
